modules/imgcodecs/src/loadsave.cpp

```diff
@@ -651,8 +651,14 @@ imdecode_( const Mat& buf, int flags, int hdrtype, Mat* mat=0 )
         if( !f )
             return 0;
         size_t bufSize = buf.cols*buf.rows*buf.elemSize();
-        fwrite( buf.ptr(), 1, bufSize, f );
-        fclose(f);
+        if( fwrite( buf.ptr(), 1, bufSize, f ) != bufSize )
+        {
+            CV_Error( CV_StsError, "failed to write image data to temporary file" );
```

Comment:
Please add `fclose(f)` call here (probably without check). `CV_Error()` doesn't return (it throws exception).

What is about to unify errors processing (see line 652 - "return 0;" statement)?